bgpd/bgp_main.c

```diff
@@ -373,7 +372,11 @@ int main(int argc, char **argv)
 		"  -n, --no_kernel    Do not install route to kernel.\n"
 		"  -S, --skip_runas   Skip capabilities checks, and changing user and group IDs.\n"
 		"  -e, --ecmp         Specify ECMP to use.\n");
-
+	/* Guard to prevent a second instance of this daemon*/
+	if (frr_guard_daemon() == FAILURE) {
```

Comment:
This is absolutely the wrong place to call this. The user can specify a different pid file as command-line option and we need to check for that.

Please do this in `frr_init()` after the `zprivs_init()` call (and don't touch any of the daemons.)